cognite/neat/_graph/transformers/_rdfpath.py

```diff
@@ -90,20 +85,45 @@ def __init__(
 
         self.limit = limit
 
-    def transform(self, graph: Graph) -> None:
-        query = self._ref_template.format(
+    def _iterate_query(self) -> str:
+        query = """SELECT DISTINCT ?subject ?object
+                            WHERE {{
+                                ?subject a <{subject_type}> .
+                                ?subject <{subject_predicate}> ?value .
+                                ?object <{object_predicate}> ?value .
+                                ?object a <{object_type}> .
+                            }}"""
+
+        if self.limit and isinstance(self.limit, int) and self.limit > 0:
+            query += f" LIMIT {self.limit}"
+
+        return query.format(
             subject_type=self.subject_type,
             subject_predicate=self.subject_predicate,
             object_type=self.object_type,
             object_predicate=self.object_predicate,
         )
 
-        if self.limit and isinstance(self.limit, int) and self.limit > 0:
-            query += f" LIMIT {self.limit}"
+    def _count_query(self) -> str:
+        query = """SELECT (COUNT(DISTINCT (?subject ?object)) as ?count)
+                    WHERE {{
+                        ?subject a <{subject_type}> .
+                        ?subject <{subject_predicate}> ?value .
+                        ?object <{object_predicate}> ?value .
+                        ?object a <{object_type}> .
+                    }}"""
+        return query.format(
+            subject_type=self.subject_type,
+            subject_predicate=self.subject_predicate,
+            object_type=self.object_type,
+            object_predicate=self.object_predicate,
+        )
+
+    def operation(self, query_result_row: ResultRow) -> RowTransformationOutput:
+        row_output = RowTransformationOutput()
 
-        triples: list[Triple] = []
-        for subject, object in graph.query(query):  # type: ignore [misc]
-            triples.append(cast(Triple, (subject, self.connection, object)))
+        subject, object = query_result_row
 
-        print(f"Found {len(triples)} connections. Adding them to the graph...")
-        add_triples_in_batch(graph, triples)
+        row_output.add_triples.append(cast(Triple, (subject, self.connection, object)))
+        row_output.instances_added_count += 1
```

Comment:
I think this is a modification and not an addition of an instance. You are adding a connection to an existing instance.